backend/tests/Unit/Infrastructure/Ai/ElephantOpenWebUiSocketSessionFactoryTest.php

```
@@ -0,0 +1,29 @@
+use App\Infrastructure\Ai\ElephantOpenWebUiSocketSessionFactory;
+use PHPUnit\Framework\TestCase;
+
+final class ElephantOpenWebUiSocketSessionFactoryTest extends TestCase
+{
+    public function testConfiguresDirectWebSocketAndTokenOnlyInAuthHandshake(): void
+    {
+        $session = (new ElephantOpenWebUiSocketSessionFactory())->create('https://example.test', 'secret-token', 3.5);
+        $property = new \ReflectionProperty($session, 'engine');
+        $engine = $property->getValue($session);
+
+        self::assertInstanceOf(ElephantOpenWebUiSocketEngine::class, $engine);
```

PHPStan (2.2.7)

[error] Call to an undefined static method Tests\Unit\Infrastructure\Ai\ElephantOpenWebUiSocketSessionFactoryTest::assertInstanceOf().


(staticMethod.notFound)